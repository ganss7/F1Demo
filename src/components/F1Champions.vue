<template>
  <div class="champions-list">
    <md-content class="heading">
      <h1>F1 Champions from {{yearFrom}} to {{yearTo}}:</h1>  
    </md-content>
    
    <md-progress-bar v-if="!championsList.length" md-mode="query"></md-progress-bar>
    <md-list>
      <transition-group
        @before-enter="beforeEnter"
        @enter="enter"
        @after-enter="afterEnter"
        @before-leave="beforeLeave"
        @leave="leave"
        @afterLeave="afterLeave"
        tag="div">
        <md-list-item v-for="(champion, index) in championsList" :key="champion.season"
          :to="{ name: 'season', params: { season: champion.season, winner: champion.DriverStandings[0].Driver } }"
          :data-index="index">
          {{index+1}}.&nbsp;<strong>{{champion.season}}</strong>&nbsp;
          -&nbsp;
          <strong>{{getDriverFullName(champion.DriverStandings[0].Driver)}}</strong>
        </md-list-item>
      </transition-group>
    </md-list>

    <!-- dialog for showing error messages -->
    <md-dialog :md-active.sync="showErrorMessage">
      <md-dialog-title>Error</md-dialog-title>
      <md-dialog-content>{{errorMessage}}</md-dialog-content>
      <md-dialog-actions>
        <md-button @click="showErrorMessage = false">Ok</md-button>
      </md-dialog-actions>
    </md-dialog>
  </div>
</template>

<script>
import f1history from '@/api/f1history'
import utils from '@/mixins/utils'
import listAnimation from '@/mixins/listAnimation'

export default {
  props: {
    yearFrom: String,
    yearTo: String
  },
  mixins: [utils, listAnimation],
  data () {
    return {
      championsList: [],
      showErrorMessage: false,
      errorMessage: ''
    }
  },
  created () {
    f1history.getChampionsList({yearFrom: this.yearFrom, yearTo: this.yearTo})
      .then(data => {
        this.championsList = data.MRData.StandingsTable.StandingsLists
        this.$options.listLength = this.championsList.length
      })
      .catch(error => {
        this.errorMessage = error.message
        this.showErrorMessage = true
      }) 
  }
}
</script>


