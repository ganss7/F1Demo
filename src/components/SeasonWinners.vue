<template>
  <div class="season-winners-list">
    <md-content class="heading">
      <h1>Season {{season}} race winners:</h1>
    </md-content>

    <md-progress-bar v-if="!winnersList.length" md-mode="query"></md-progress-bar>
    <md-list>
      <transition-group
        @before-enter="beforeEnter"
        @enter="enter"
        @after-enter="afterEnter"
        @before-leave="beforeLeave"
        @leave="leave"
        @afterLeave="afterLeave"
        tag="div">
        <md-list-item v-for="(winner, index) in winnersList" :key="winner.round"
          :class="`${myWinner &&  myWinner.driverId === winner.Results[0].Driver.driverId ? 'highlight' : ''}`"
          :data-index="index">
          {{index+1}}.&nbsp;<strong>{{winner.raceName}}</strong>
          &nbsp;-&nbsp;
          <a :href="winner.Results[0].Driver.url" target="_blank">{{getDriverShortName(winner.Results[0].Driver)}}</a>
        </md-list-item>
      </transition-group>
    </md-list>

    <!-- dialog for showing error messages -->
    <md-dialog :md-active.sync="showErrorMessage">
      <md-dialog-title>Error</md-dialog-title>
      <md-dialog-content>{{errorMessage}}</md-dialog-content>
      <md-dialog-actions>
        <md-button @click="showErrorMessage = false">Ok</md-button>
      </md-dialog-actions>
    </md-dialog>
  </div>
</template>

<script>
import f1history from '@/api/f1history'
import utils from '@/mixins/utils'
import listAnimation from '@/mixins/listAnimation'

export default {
  props: {
    season: String,
    winner: Object
  },
  mixins: [utils, listAnimation],
  data () {
    return {
      winnersList: [],
      myWinner: this.winner,
      showErrorMessage: false,
      errorMessage: ''
    }
  },
  created () {
    f1history.getSeasonWinnersList(this.season)
      .then(data => this.winnersList = data.MRData.RaceTable.Races)
      .catch(error => {
        this.errorMessage = error.message
        this.showErrorMessage = true
      }) 
    
    if (!this.myWinner) {
      f1history.getSeasonWinner(this.season)
        .then(data => this.myWinner = data.MRData.StandingsTable.StandingsLists[0].DriverStandings[0].Driver)
        .catch(error => {
          this.errorMessage = error.message
          this.showErrorMessage = true
        })
    }
  }
}
</script>

<style lang="scss" scoped>
  #app .md-list-item.highlight {
    background: var(--md-theme-default-accent);
  }
</style>

