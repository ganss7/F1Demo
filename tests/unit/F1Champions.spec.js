import { shallowMount, createLocalVue  } from '@vue/test-utils'

import { MdApp, MdToolbar, MdContent, MdList, MdDialog, MdButton, MdProgress } from 'vue-material/dist/components'
import F1Champions from '@/components/F1Champions.vue'

const localVue = createLocalVue()

localVue.use(MdApp)
localVue.use(MdToolbar)
localVue.use(MdContent)
localVue.use(MdList)
localVue.use(MdDialog)
localVue.use(MdButton)
localVue.use(MdProgress)

jest.mock('@/api/f1history')

const inputProps = {
  yearFrom: '2005',
  yearTo: '2015'
}

describe('F1Champions.vue', () => {

  it('renders correctrly', (done) => {

    const wrapper = shallowMount(F1Champions, {
      localVue,
      propsData: inputProps
    })

    wrapper.vm.$nextTick(() => {
      expect(wrapper.element).toMatchSnapshot()
      done()
    })

  })

})
