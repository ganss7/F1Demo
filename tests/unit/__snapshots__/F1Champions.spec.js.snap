// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`F1Champions.vue renders correctrly 1`] = `
<div
  class="champions-list"
>
  <mdcontent-stub
    class="heading"
    mdtag="div"
  >
    <h1>
      F1 Champions from 2005 to 2015:
    </h1>
  </mdcontent-stub>
   
  <!---->
   
  <mdlist-stub>
    <span
      tag="div"
    >
      <mdlistitem-stub
        data-index="0"
        dataindex="0"
        to="[object Object]"
      >
        
        1. 
        <strong>
          2005
        </strong>
         
        - 
        
        <strong>
          Fernando Alonso
        </strong>
      </mdlistitem-stub>
      <mdlistitem-stub
        data-index="1"
        dataindex="1"
        to="[object Object]"
      >
        
        2. 
        <strong>
          2006
        </strong>
         
        - 
        
        <strong>
          Fernando Alonso
        </strong>
      </mdlistitem-stub>
      <mdlistitem-stub
        data-index="2"
        dataindex="2"
        to="[object Object]"
      >
        
        3. 
        <strong>
          2007
        </strong>
         
        - 
        
        <strong>
          Kimi Räikkönen
        </strong>
      </mdlistitem-stub>
      <mdlistitem-stub
        data-index="3"
        dataindex="3"
        to="[object Object]"
      >
        
        4. 
        <strong>
          2008
        </strong>
         
        - 
        
        <strong>
          Lewis Hamilton
        </strong>
      </mdlistitem-stub>
      <mdlistitem-stub
        data-index="4"
        dataindex="4"
        to="[object Object]"
      >
        
        5. 
        <strong>
          2009
        </strong>
         
        - 
        
        <strong>
          Jenson Button
        </strong>
      </mdlistitem-stub>
      <mdlistitem-stub
        data-index="5"
        dataindex="5"
        to="[object Object]"
      >
        
        6. 
        <strong>
          2010
        </strong>
         
        - 
        
        <strong>
          Sebastian Vettel
        </strong>
      </mdlistitem-stub>
      <mdlistitem-stub
        data-index="6"
        dataindex="6"
        to="[object Object]"
      >
        
        7. 
        <strong>
          2011
        </strong>
         
        - 
        
        <strong>
          Sebastian Vettel
        </strong>
      </mdlistitem-stub>
      <mdlistitem-stub
        data-index="7"
        dataindex="7"
        to="[object Object]"
      >
        
        8. 
        <strong>
          2012
        </strong>
         
        - 
        
        <strong>
          Sebastian Vettel
        </strong>
      </mdlistitem-stub>
      <mdlistitem-stub
        data-index="8"
        dataindex="8"
        to="[object Object]"
      >
        
        9. 
        <strong>
          2013
        </strong>
         
        - 
        
        <strong>
          Sebastian Vettel
        </strong>
      </mdlistitem-stub>
      <mdlistitem-stub
        data-index="9"
        dataindex="9"
        to="[object Object]"
      >
        
        10. 
        <strong>
          2014
        </strong>
         
        - 
        
        <strong>
          Lewis Hamilton
        </strong>
      </mdlistitem-stub>
      <mdlistitem-stub
        data-index="10"
        dataindex="10"
        to="[object Object]"
      >
        
        11. 
        <strong>
          2015
        </strong>
         
        - 
        
        <strong>
          Lewis Hamilton
        </strong>
      </mdlistitem-stub>
    </span>
  </mdlist-stub>
   
  <mddialog-stub
    mdbackdrop="true"
    mdbackdropclass="md-dialog-overlay"
    mdclickoutsidetoclose="true"
    mdcloseonesc="true"
    mdfullscreen="true"
  >
    <mddialogtitle-stub>
      Error
    </mddialogtitle-stub>
     
    <mddialogcontent-stub>
      
    </mddialogcontent-stub>
     
    <mddialogactions-stub>
      <mdbutton-stub
        type="button"
      >
        Ok
      </mdbutton-stub>
    </mddialogactions-stub>
  </mddialog-stub>
</div>
`;
